reconcile/utils/aws_api_typed/s3.py

```diff
@@ -0,0 +1,25 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3 import S3Client
+    from mypy_boto3_s3.literals import BucketLocationConstraintType
+else:
+    S3Client = BucketLocationConstraintType = object
+
+
+class AWSApiS3:
+    def __init__(self, client: S3Client) -> None:
+        self.client = client
+
+    def create_bucket(self, name: str, region: str) -> str:
+        """Create an S3 bucket without any ACLs therefore the creator will be the owner and returns the ARN."""
+        bucket_kwargs = {}
+        if region != "us-east-1":
+            # you can't specify the location if it's us-east-1 :(
```

Comment:
maybe attach the doc link here https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateBucketConfiguration.html, as this is an unusual logic.

Reply:
done in [b3c93bf](https://github.com/app-sre/qontract-reconcile/pull/4253/commits/b3c93bf03e7ae4eb97516f2cc236574a37b49985)